src/passes/transform/fold_constants.cpp

```diff
@@ -374,8 +374,103 @@ namespace CoreIR {
                        instPT->sel("in")->sel("out"));
           inlineInstance(instPT);
             
+        } else if (sigValue0.has_value()) {
+          BitVec sigVal0 = sigValue0.get_value();
+
+          if ((sigVal0.bitLength() == 1) && (sigVal0 == BitVec(1, 0))) {
+            cout << "Should remove and with 0" << endl;
```

Comment:
What is this comment? If it should error, then you should use ASSERT, otherwise there should be nothing to cout